src/instructlab/utils.py

```diff
@@ -164,14 +164,18 @@ def lab_check_callback(*args, **kwargs):
 """Taxonomy folders which are also the schema names"""
 
 
-def istaxonomyfile(fn):
+def is_taxonomy_file(fn: str):
     path = Path(fn)
+    if path.suffix == ".yml" and path.parts[0] in TAXONOMY_FOLDERS:
+        logger.warning(
+            f"Found a '.yml' file: {path}: taxonomy files must have a '.yaml' extension. File will not be checked."
+        )
     return path.suffix == ".yaml" and path.parts[0] in TAXONOMY_FOLDERS
```

Comment:
should we just log when this returns False?.. (Plus adding the validation for `qna.yaml` full name, as @bjhargrave suggested above.)

Reply:
It can return false for attribution.txt which should not log a message.

Reply:
Right, and `attribution.txt` is a valid file - this particular function is only used to determine whether or not a file pointer is externally recognizable as a possible taxonomy entry - additional functions beyond this do checks of
1. Byte content (attempt loading and check if the content is not empty and matches the generic `Mapping` type)
2. Generic YAML linting 
3. Specific InstructLab Schema Validation

Reply:
I added a commit to explain this more in-line in the code

Reply:
We also will need to allow md files in the folder as someone could refer to a prior commit in their taxonomy repo for knowledge documents. Since the folder can contain other files than the qna, this is why the qna.yaml file name is important.